modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js

```diff
@@ -1835,6 +1835,34 @@ return view.extend({
 						o.placeholder = '201';
 						o.rmempty = true;
 
+						if (L.hasSystemFeature('hostapd', 'ocv') || L.hasSystemFeature('wpasupplicant', 'ocv')) {
+							o = ss.taboption('encryption', form.ListValue, 'ocv', _('Operating Channel Validation'), _("Note: Workaround mode allows a STA that claims OCV capability to connect even if the STA doesn't send OCI or negotiate PMF."));
+							o.value('0', _('Disabled'));
+							o.value('1', _('Enabled'));
+							o.value('2', _('Enabled (workaround mode)'));
+
+							add_dependency_permutations(o, { mode: ['ap', 'ap-wds', 'sta', 'sta-wds'], encryption: ['psk-mixed', 'psk2', 'sae', 'sae-mixed', 'wpa2', 'wpa3', 'wpa3-mixed', 'owe'] });
+
+							o.validate = function(section_id, value) {
+								var mfpopt = this.section.children.filter(function(o) { return o.option == 'ieee80211w' })[0],
+								mfpval = mfpopt.formvalue(section_id);
+
+								if ((value == '1' || value == '2') && mfpval == '0') {
+									return _('Requires 802.11w to be set to optional or required.');
+								}
+
+								var modeopt = this.section.children.filter(function(o) { return o.option == 'mode' })[0],
+								modeval = modeopt.formvalue(section_id),
+								modetitle = modeopt.vallist[modeopt.keylist.indexOf(modeval)];
+
+								if ((value == '2') && ((modeval == 'sta') || (modeval == 'sta-wds'))) {
```

Comment:
Same here for depends

Reply:
I can't really get around having a validator function here because I need to notify the user that the workaround mode will not work in client mode.

Reply:
So in other words, in what modes *will* it work?

depends also has available: `!contains` and `!reverse`, and the following:
` opt.depends("test.section1.foo", "bar")`
Require the "foo" form option within the "section1" section to be set to "bar". 

so it seems like one can instead do:
```js

							o.depends({ieee80211w: '1', ...});
							o.depends({ieee80211w: '2', ...});
```
where `...` is the necessary magic of "and `mode` is not sta*"

Reply:
Workaround mode will only work in `sta` and `sta-wds` mode. Just to be clear, do you want me to get rid of the validator function altogether? I might very well be missing something here, but I don't think it's possible to use depends to make individual options in the drop-down menu depend on some other value. Depends causes the entire ocv option to appear or be hidden based on whether the dependencies are fulfilled. That means that I would have to create two different taboption instances for ocv, with one for when mode is either `ap` or `ap-wds`, and the other for when mode is `sta` or `sta-wds`. The question then is how to deal with the info text about workaround mode. Do I only show it when mode is set to `sta` or `sta-wds`? Seems a bit messy, but I guess that could work.

Reply:
> Workaround mode will only work in `sta` and `sta-wds` mode. 

Now we're getting somewhere. 

```js
							//this means: 
							o.depends('ieee80211w', '1'); // 'ocv' depends on 11w enabled... 
							o.depends('ieee80211w', '2'); // OR 'ocv' depends on 11w compat mode
```

We need:

```js
							o.depends('ieee80211w', '1'); // 'ocv' depends on 11w enabled... 
							o.depends({ieee80211w: '2', ...}); // OR 'ocv' depends on 11w compat mode AND *Workaround mode will only work in `sta` and `sta-wds` mode.*
```

Reply:
Sorry, I think I got you confused with the workaround mode stuff. 802.11w has two possible "enabled" values:

```
1 - optional
2 - required
```

As long as ieee80211w is set to one of these values ocv will work in both ap (`ap`, `ap-wds`) and client (`sta`, `sta-wds`) modes. That means we can't account for mode having to be either `sta` or `sta-wds` when using ocv workaround mode with the help of depends. So, we're back to either keeping the validator function (which we don't want to because of the translation issues you pointed out), or to have two different drop-down lists depending on whether ap or client modes have been selected.

Reply:
> That means we can't account for mode having to be either `sta` or `sta-wds` when using ocv workaround mode with the help of depends.


We *can't*?

https://github.com/openwrt/luci/blob/e2a106db27a0b45983aed3f80b12bead2673fd93/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js#L1695

https://github.com/openwrt/luci/blob/e2a106db27a0b45983aed3f80b12bead2673fd93/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js#L1007

Reply:
Sure, those can show/hide the entire dropdown list based on whether the dependencies are fulfilled or not, but that doesn't help us in this case. As far as I can see, we can't show/hide individual options in the dropdown list by using depends. If that was possible then we could simply have shown the "workaround mode" option when mode was set to `sta` or `sta-wds`.

Reply:
Just realised that I said that workaround mode will only work in `sta` and `sta-wds` mode, when it's actually in `ap` and `ap-wds` mode. Sorry about that!

Reply:
Well, if you want to do it that way, gate the `o.value('2')` behind an `if` condition to prevent even showing it. 

We should probably also default the setting (when it is not already set) to 0.

Reply:
I couldn't figure out how to do that, so I decided to keep the validator function and change the string so that it's static and doesn't contain values that can change. Is that ok with regard the the translation issue you pointed out earlier? I've also set the default value to 0.

Reply:
try this?
```js
...
							var modeopt = this.section.children.filter(function(o) { return o.option == 'mode' })[0],
							modeval = modeopt.formvalue(section_id);
							if (!((modeval == 'sta') || (modeval == 'sta-wds'))) {
								o.value('2', _('Enabled (workaround mode)'));
							}
							o.depends('ieee80211w', '1');
							o.depends('ieee80211w', '2');

```

Reply:
That doesn't work because I just end up with a "this is undefined" error. I assume I need to put this inside some method or something? But right now I'm just wondering what the issue is with the PR at this point. Is it not acceptable as is?

Reply:
Largely yes. But if we can prevent making users feel dumb or that they did something wrong, and having them try to figure out what's wrong, it's best to prevent those cases altogether (preventing workaround mode when not in sta* modes).

Reply:
Fair enough. I'll try to look at it some more to see if I can make that work.

Reply:
I think my js knowledge is just too limited to be able to figure out how to be able to get the formvalue of mode. I need that for the if-statement that adds the workaround mode value. The code works fine inside the validation method, but when I bring it outside like you suggested below I get the "this is undefined" error.

> try this?
> 
> ```js
> ...
> 							var modeopt = this.section.children.filter(function(o) { return o.option == 'mode' })[0],
> 							modeval = modeopt.formvalue(section_id);
> 							if (!((modeval == 'sta') || (modeval == 'sta-wds'))) {
> 								o.value('2', _('Enabled (workaround mode)'));
> 							}
> 							o.depends('ieee80211w', '1');
> 							o.depends('ieee80211w', '2');
> ```

Very naively, I decided to try to replace `this` with `o`, and I get the `section_id` from `o.section.section` (not sure if this is correct?):

```javascript
var modeopt = o.section.children.filter(function(o) { return o.option == 'mode' })[0],
section_id = o.section.section,
modeval = modeopt.formvalue(section_id);
if (!((modeval == 'sta') || (modeval == 'sta-wds'))) {
    o.value('2', _('Enabled (workaround mode)'));
}
```

But trying to get the formvalue just ends up giving me a "this.root is undefined" error.

Reply:
```js
						var mode = uci.get('wireless', section_id, 'mode');
```
? Although `this` should work. Perhaps you just need to clear your cache and reload the page?

Reply:
While `uci.get` is able to return the current value in the config, I don't think it's optimal because it's not reading the value that's currently selected in Luci. A user would be able to change the mode to `sta` or `sta-wds` and without saving open the security tab and then set operating channel validation to `Enabled (workaround mode)`.

Also, I tried refreshing while ignoring cached content (ctrl-shift-r) and `this` still didn't work. I guess it's because the script uses strict mode: _When invoked as a standalone function (not attached to an object: func()), this typically refers to the global object (in non-strict mode) or undefined (in strict mode)._[1]

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this

Reply:
Ah yeah, the in-memory version. Then your current version looks most appropriate.

Reply:
Can this be merged now?